libmate-desktop/mate-bg-crossfade.c

```diff
@@ -528,18 +679,37 @@ mate_bg_crossfade_start (MateBGCrossfade *fade,
 	fade->priv->timeout_id = g_source_attach (source, context);
 	g_source_unref (source);
 
-	fade->priv->window = window;
-	cairo_pattern_t *pattern;
-	pattern = cairo_pattern_create_for_surface (fade->priv->fading_surface);
-	gdk_window_set_background_pattern (fade->priv->window, pattern);
-	cairo_pattern_destroy (pattern);
-	draw_background (fade);
-
 	fade->priv->is_first_frame = TRUE;
 	fade->priv->total_duration = .75;
 	fade->priv->start_time = get_current_time ();
 }
 
+/**
+ * mate_bg_crossfade_start_widget:
+ * @fade: a #MateBGCrossfade
+ * @widget: The #GtkWidget to draw crossfade on
+ *
+ * This function initiates a quick crossfade between two surfaces on
+ * the background of @widget.  Before initiating the crossfade both
+ * mate_bg_crossfade_start() and mate_bg_crossfade_end() need to
```

Comment:
Minor issue: looks like ```mate_bg_crossfade_set_start_surface``` and ```mate_bg_crossfade_set_end_surface``` should be meant here instead? (Same as in the function above - where it was copied from)